dev/ProgressRing/ProgressRing.cpp

```diff
@@ -147,28 +236,59 @@ void ProgressRing::SetLottieBackgroundColor(winrt::impl::com_ref<AnimatedVisuals
         }
         else
         {
-            // Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
+             //Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
             return SharedHelpers::FindInApplicationResources(s_DefaultBackgroundThemeResourceName).as<winrt::SolidColorBrush>().Color();
         }
     }();
 
-    progressRingIndeterminate->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    if (const auto progressRingAnimation = animatedVisualSource.try_as<AnimatedVisuals::ProgressRingIndeterminate>())
+    {
+        progressRingAnimation->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    }
+}
+
+void ProgressRing::UpdateLottieProgress()
+{
+    if (auto&& player = m_player.get())
+    {
+        const double value = Value();
+        const double range = Maximum() - Minimum();
+        const double fromProgress = m_oldValue / range;
+        const double toProgress = value / range;
+
+        const auto _ = player.PlayAsync(fromProgress, toProgress, false);
+        m_oldValue = value;
```

Comment:
>m_oldValue [](start = 8, length = 10)

doesn't this mean that we wont update the oldValue when the AVP plays? so if we play the animation from 0% to 50% then update a range based property causing UpdateLottieProgress to fire, the OldValue would be 0% not 50%?

Reply:
The `m_oldValue` stores the 'current' value to be used as the `fromProgress`  value the next time RangeBase is updated. So in the scenario above, `m_oldValue` would be 50%.

Reply:
I don't get how m_oldValue is getting set in the middle of the animation but if it is then this is fine.

---
In reply to: [453820440](https://github.com/microsoft/microsoft-ui-xaml/pull/2584#discussion_r453820440) [](ancestors = 453820440)